index.js

```diff
@@ -32,7 +33,26 @@ module.exports = function(content) {
 	if (typeof config.filters === 'object') {
 		Engine.registerFilters(config.filters);
 	}
-	return Engine.parseAndRender(content, config.data || {}).then(function(
+
+	// Get template data from data-file of loader options
+	let templateData = {};
+
+	if (config.data) {
+		templateData = config.data;
+	} else {
+		const dataPath = config.dataPath || 'data';
```

Comment:
Good job! But I suggest not to introduce new options, but to make it possible to pass either an object or a function to "data", and load json in the webpack config.

Reply:
Yes! I thought of that once I submitted the pull request. I think that I like that approach the most too.
Gonna be working on it this week 👍 and let you know once it is done.

Reply:
There it goes, much clean.